### PR TITLE
Support CCS query

### PR DESCRIPTION
**Pull request by zeeev**
This would generalize to all CCS query, but the test data hit the known bug, so I just left it for by-strand ccs.

### CODE REVIEW

File: src/AlignmentResult.cpp [hunk 1]
```diff
@@ -154,7 +154,7 @@ std::ostream& operator<<(std::ostream& out, const AlignmentResult& a)
 }
 
 BAM::BamRecord AlnToBam(const int32_t refId, const BAM::BamHeader& header,
-                        const AlignmentResult& aln, const BAM::BamRecord& read)
+                        const AlignmentResult& aln, const BAM::BamRecord& read, bool ccs)
```
Comment (armintoepfer) [inline src/AlignmentResult.cpp:157]:
```suggestion
                        const AlignmentResult& aln, const BAM::BamRecord& read, const bool ccs)
```

File: src/AlignmentResult.cpp [hunk 2]
```diff
@@ -168,12 +168,14 @@ BAM::BamRecord AlnToBam(const int32_t refId, const BAM::BamHeader& header,
     record.Map(refId, aln.rStart, aln.rReversed ? BAM::Strand::REVERSE : BAM::Strand::FORWARD,
                BAM::Cigar::FromStdString(cigarStr), aln.mapq);
 
+    int32_t readstart = ccs ? 0 : read.QueryStart();
```
Comment (armintoepfer) [inline src/AlignmentResult.cpp:171]:
```suggestion
    const int32_t readstart = ccs ? 0 : read.QueryStart();
```

File: src/main.cpp [hunk 3]
```diff
@@ -371,7 +386,9 @@ int RunnerSubroutine(const CLI_v2::Results& options)
                                                 std::ref(workQueue), std::ref(writer), numCcsReads);
 
     const auto Submit = [&header](const std::vector<BAM::BamRecord>& clrRecords,
-                                  const BAM::BamRecord& ccsRecord, const int32_t curCcsIdx) {
+                                  const BAM::BamRecord& ccsRecord, const int32_t curCcsIdx,
+                                  bool ccs) {
```
Comment (armintoepfer) [inline src/main.cpp:390]:
```suggestion
                                  const bool ccs) {
```

File: src/main.cpp [hunk 4]
```diff
@@ -371,7 +386,9 @@ int RunnerSubroutine(const CLI_v2::Results& options)
...
+        PBLOG_DEBUG << "About to align!";
```
Comment (armintoepfer) [inline src/main.cpp:391]:
```suggestion
```

File: src/main.cpp [hunk 5]
```diff
@@ -380,11 +397,12 @@ int RunnerSubroutine(const CLI_v2::Results& options)
             for (const auto& a : aln) {
                 if (a->isAligned) {
                     alnRecords.emplace_back(
-                        AlnToBam(curCcsIdx, header, *a, clrRecords[subreadIdx]));
+                        AlnToBam(curCcsIdx, header, *a, clrRecords[subreadIdx], ccs));
                 }
             }
             ++subreadIdx;
         }
+        PBLOG_DEBUG << "Aligned!";
```
Comment (armintoepfer) [inline src/main.cpp:405]:
```suggestion
```

File: src/main.cpp [hunk 6]
```diff
@@ -408,9 +433,15 @@ int RunnerSubroutine(const CLI_v2::Results& options)
             PBLOG_DEBUG << "CLR : " << clrRecord.FullName();
             clrRecords.emplace_back(clrRecord);
             GetNextRecord();
+            // TODO this is required because the do while loop has a known bug:
+            //   If the query bam contains only one hole, the loop goes inf. The by-strand case is one hole.
+            if (settings.byStrand) {
+                if (clrRecords.size() > 1) break;
```
Comment (armintoepfer) [inline src/main.cpp:439]:
```suggestion
                if (clrRecords.size() > 1) {
                    break;
                }
```

Okay, but I'll fix it properly after this PR

**Review by armintoepfer — CHANGES_REQUESTED** [review]

**Comment by armintoepfer** [comment]
Please squash your commits before merging

**Review by armintoepfer — APPROVED** [review]

**Comment by zeeev** [comment]
Thanks @armintoepfer

**Pull request merged**